lib/stewfinder.rb

```diff
@@ -1,5 +1,64 @@
 require 'stewfinder/version'
+require 'yaml'
 
-module Stewfinder
-  # Your code goes here...
+class Stewfinder
+  def initialize(filename)
+    @name = filename
+  end
+
+  def find
+    stewards = get_stewards
+
+    puts 'Stewards for this file:'
+    if stewards.empty?
+      puts 'None'
+    else
+      puts ' - ' + stewards.uniq.sort.join("\n - ")
+    end
+  end
+
+  private
+
+  def get_stewfiles
+    filename = @name.split('/')
+    stewfiles = []
+    while filename.size > 1
+      filename.delete_at(-1)
+      files = `ls #{filename.join('/')}`.split("\n").inspect
```

Comment:
Because it's backticks, `ls` itself sends some error messaging out to the console and returns `nil` to the script.

```
ls: /Users/alexanderstandke/git/apm_bundle/apps/property/engines/saffron/app/views/saffrasdfon/patterns: No such file or directory
ls: /Users/alexanderstandke/git/apm_bundle/apps/property/engines/saffron/app/views/saffrasdfon: No such file or directory
Stewards for this file:
 - XanderStrike
 - aaronpanch
 - dtognazzini
 - jonkessler
 - rmacklin
 - scuendet